pkg/cloudproviders/aws/utils/utils.go

```diff
@@ -0,0 +1,52 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"regexp"
+
+	v1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/ptr"
+
+	"github.com/aws/karpenter/pkg/operator/injection"
+)
+
+// ParseInstanceID parses the provider ID stored on the node to get the instance ID
+// associated with a node
+func ParseInstanceID(node *v1.Node) (*string, error) {
```

Comment:
Was this a bug with our code? Why is this code different than the previous?

Reply:
This is an improvement on the last method, which wasn't super crisp. It's moving into utils as a readying mechanism for the NTH PR which also utilizes it